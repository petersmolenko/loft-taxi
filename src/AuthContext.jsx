import React, { useState } from "react";
import { Grid, Typography } from "@material-ui/core";

const AuthContext = React.createContext(null);

const AuthProvider = (props) => {
    const [isLoggedIn, setIsLoggedIn] = useState(false);
    const login = (email, password) => {
        setIsLoggedIn(true);
        console.log(`User ${email} is authorized.`);
    };
    const logout = () => setIsLoggedIn(false);
    const { children } = props;

    return (
        <AuthContext.Provider value={{ isLoggedIn, login, logout }}>
            {children}
        </AuthContext.Provider>
    );
};

const AuthHOC = (WrappedComponent) => (props) => (
    <AuthContext.Consumer>
        {(contextProps) => (
            <>
                {contextProps.isLoggedIn ? (
                    <WrappedComponent {...contextProps} {...props} />
                ) : (
                    <Grid
                        item
                        container
                        justify="center"
                        alignItems="center"
                        style={{ flexGrow: 1 }}
                    >
                        <Typography
                            variant="h3"
                            component="h3"
                            style={{ color: "#ffc617" }}
                        >
                            The user is not logged in:(
                        </Typography>
                    </Grid>
                )}
            </>
        )}
    </AuthContext.Consumer>
);

export { AuthContext, AuthProvider, AuthHOC };
