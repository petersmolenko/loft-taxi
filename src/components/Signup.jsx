import React, { useState } from "react";
import { signUp } from "../redux/modules/auth";
import { connect } from "react-redux";
import {
    Grid,
    Paper,
    Typography,
    Link,
    TextField,
    Button,
} from "@material-ui/core";
import logo from "../assets/logo-white.svg";
import PropTypes from "prop-types";
import { Link as LinkRoute } from "react-router-dom";

const Signup = ({ signup }) => {
    const [regData, setRegData] = useState({
        email: "",
        password: "",
        name: "",
        surname: "",
    });

    return (
        <>
            <Grid item xs={4} lg={3}>
                <img className="MainLogo" src={logo} alt="" />
            </Grid>
            <Grid item xs={6} lg={5}>
                <Paper elevation={0} className="AppForm">
                    <Typography
                        variant="h4"
                        component="h4"
                        className="AppForm_margin"
                    >
                        Регистрация
                    </Typography>
                    <Typography
                        variant="body1"
                        component="p"
                        className="AppForm_margin"
                    >
                        Уже зарегистрированы?{" "}
                        <Link
                            href="/"
                            component={LinkRoute}
                            to="/login"
                            color="primary"
                        >
                            Войти
                        </Link>
                    </Typography>

                    <form noValidate autoComplete="off">
                        <Grid container justify="center" direction="column">
                            <Grid item>
                                <TextField
                                    required
                                    id="userEmail"
                                    label="Адрес электронной почты"
                                    className="AppForm_margin"
                                    inputProps={{
                                        "data-testid": "userEmailField",
                                    }}
                                    type="email"
                                    onChange={(e) =>
                                        setRegData({
                                            ...regData,
                                            email: e.target.value,
                                        })
                                    }
                                    fullWidth
                                />
                            </Grid>
                            <Grid
                                item
                                className="AppForm_margin"
                                justify="space-between"
                                container
                            >
                                <TextField
                                    required
                                    id="userFirstName"
                                    inputProps={{
                                        "data-testid": "userFirstNameField",
                                    }}
                                    label="Имя"
                                    onChange={(e) =>
                                        setRegData({
                                            ...regData,
                                            name: e.target.value,
                                        })
                                    }
                                    type="email"
                                />
                                <TextField
                                    required
                                    inputProps={{
                                        "data-testid": "userSurenameField",
                                    }}
                                    id="userSureName"
                                    label="Фамилия"
                                    onChange={(e) =>
                                        setRegData({
                                            ...regData,
                                            surname: e.target.value,
                                        })
                                    }
                                    type="email"
                                />
                            </Grid>
                            <Grid item>
                                <TextField
                                    required
                                    id="userPassword"
                                    inputProps={{
                                        "data-testid": "userPasswordField",
                                    }}
                                    label="Пароль"
                                    type="password"
                                    onChange={(e) =>
                                        setRegData({
                                            ...regData,
                                            password: e.target.value,
                                        })
                                    }
                                    className="AppForm_margin"
                                    fullWidth
                                />
                            </Grid>
                            <Grid item align="right">
                                <Button
                                    variant="contained"
                                    color="primary"
                                    style={{
                                        background: "#ffc617",
                                        color: "rgba(0, 0, 0, 0.87)",
                                    }}
                                    component={LinkRoute}
                                    to="/"
                                    onClick={() => {
                                        signup(regData);
                                    }}
                                >
                                    Зарегистрироваться
                                </Button>
                            </Grid>
                        </Grid>
                    </form>
                </Paper>
            </Grid>
        </>
    );
};

Signup.propTypes = {
    signup: PropTypes.func,
};

const mapStateToProps = null;
const mapDispatchToProps = (dispatch) => ({
    signup: (auth) => dispatch(signUp(auth)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Signup);
