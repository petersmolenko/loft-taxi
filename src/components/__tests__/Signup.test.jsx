import React from "react";
import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { AuthProvider } from "../../AuthContext";
import Signup from "../Signup";

describe("Login testing", () => {
    it("renders correctly", () => {
        const handler = jest.fn();
        const tree = renderer
            .create(
                <AuthProvider>
                    <Signup setRoute={handler} />
                </AuthProvider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("login-link click", () => {
        const handler = jest.fn();
        const { getByText } = render(
            <AuthProvider>
                <Signup setRoute={handler} />
            </AuthProvider>
        );
        fireEvent.click(getByText("Войти"));
        expect(handler.mock.calls[0][0]).toBe("login");
    });

    it("signup-button click", () => {
        const handler = jest.fn();
        const { getByText } = render(
            <AuthProvider>
                <Signup setRoute={handler} />
            </AuthProvider>
        );
        fireEvent.click(getByText("Зарегистрироваться"));
        expect(handler.mock.calls[0][0]).toBe("map");
    });
    it("change form fields", () => {
        const handler = jest.fn();
        const { getByTestId } = render(
            <AuthProvider>
                <Signup setRoute={handler} />
            </AuthProvider>
        );

        fireEvent.click(getByTestId("userEmailField"), {
            target: { value: "email" },
        });
        expect(getByTestId("userEmailField").value).toBe("email");
        fireEvent.click(getByTestId("userFirstNameField"), {
            target: { value: "firstname" },
        });
        expect(getByTestId("userFirstNameField").value).toBe("firstname");
        fireEvent.click(getByTestId("userSurenameField"), {
            target: { value: "surename" },
        });
        expect(getByTestId("userSurenameField").value).toBe("surename");
        fireEvent.click(getByTestId("userPasswordField"), {
            target: { value: "password" },
        });
        expect(getByTestId("userPasswordField").value).toBe("password");
    });
});
