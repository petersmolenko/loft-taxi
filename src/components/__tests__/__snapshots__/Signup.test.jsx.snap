// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login testing renders correctly 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-lg-3"
  >
    <img
      alt=""
      className="MainLogo"
      src="logo-white.svg"
    />
  </div>,
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-5"
  >
    <div
      className="MuiPaper-root AppForm MuiPaper-elevation0 MuiPaper-rounded"
    >
      <h4
        className="MuiTypography-root AppForm_margin MuiTypography-h4"
      >
        Регистрация
      </h4>
      <p
        className="MuiTypography-root AppForm_margin MuiTypography-body1"
      >
        Уже зарегистрированы?
         
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Войти
        </a>
      </p>
      <form
        autoComplete="off"
        noValidate={true}
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root AppForm_margin MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                data-shrink={false}
                htmlFor="userEmail"
                id="userEmail-label"
              >
                Адрес электронной почты
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  data-testid="userEmailField"
                  disabled={false}
                  id="userEmail"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="email"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root AppForm_margin MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                data-shrink={false}
                htmlFor="userFirstName"
                id="userFirstName-label"
              >
                Имя
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  data-testid="userFirstNameField"
                  disabled={false}
                  id="userFirstName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="email"
                />
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                data-shrink={false}
                htmlFor="userSureName"
                id="userSureName-label"
              >
                Фамилия
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  data-testid="userSurenameField"
                  disabled={false}
                  id="userSureName"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="email"
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="MuiFormControl-root MuiTextField-root AppForm_margin MuiFormControl-fullWidth"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
                data-shrink={false}
                htmlFor="userPassword"
                id="userPassword-label"
              >
                Пароль
                <span
                  className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  data-testid="userPasswordField"
                  disabled={false}
                  id="userPassword"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="password"
                />
              </div>
            </div>
          </div>
          <div
            align="right"
            className="MuiGrid-root MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "background": "#ffc617",
                  "color": "rgba(0, 0, 0, 0.87)",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Зарегистрироваться
              </span>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
]
`;
